### ISSUE TITLE
Add Poynting components to plot_fields

### ISSUE THREAD

**Issue by thomasdorch** [NanoComp/meep#2267]
Currently, visualization.plot_fields accepts one of:
```
[mp.Ex, mp.Ey, mp.Ez, mp.Er, mp.Ep, mp.Dx, mp.Dy, mp.Dz, mp.Hx, mp.Hy, mp.Hz]
```
as valid arguments to `fields`. Can we add
```
[mp.Sx, mp.Sy, mp.Sz, mp.Sr, mp.Sp]
```
to this list as well?

**Comment by smartalecH** [comment]
Keep in mind that the poynting flux is an integral over space and is a scalar output. So there's not really anything to visualize. 

You can view things like $E^2$ using function modifiers (as described in the docs).

**Comment by smartalecH** [comment]
Oh I see you are talking about the "derived components" of the Poynting flux, as we call them. 

Sure we can add them to the list. Want to add it to your PR? Might as well add all the other components too. 

(Typically we break things down as granularity as possible, meaning different PRs for different features, but you might as well lump this one in with your other PR)

**Comment by thomasdorch** [comment]
Oh I see how that was unclear, the components were what I was referring to! 

I was thinking of creating another PR for some more type hinting and a fix (if appropriate) for #2268. That way #2264 is more specific to just fixing what's broken. How does that sound?
